src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -61,12 +62,30 @@ export type TopNavMenuProps = StatefulSearchBarProps &
  **/
 
 export function TopNavMenu(props: TopNavMenuProps): ReactElement | null {
-  const { config, showSearchBar, ...searchBarProps } = props;
+  const { config, badges, showSearchBar, ...searchBarProps } = props;
 
   if ((!config || config.length === 0) && (!showSearchBar || !props.data)) {
     return null;
   }
 
+  function renderBadges(): ReactElement | null {
+    if (!badges || badges.length === 0) return null;
+    return (
+      <span className={'kbnTopNavMenu__badgeGroup'}>
+        <EuiBadgeGroup>
```

Comment:
I only did some quick Inspector hacking, but I think you can just move the `className` onto the `EuiBadgeGroup` and remove the span wrapper.

Reply:
Yes, good call!